hpack/src/main/java/com/twitter/hpack/HeaderField.java

```diff
@@ -30,15 +30,38 @@ static int sizeOf(byte[] name, byte[] value) {
   }
 
   final byte[] name;
+  final String nameString;
   final byte[] value;
 
+  // Used to store an annotation with the header entry
+  Object valueAnnotation;
+
   // This constructor can only be used if name and value are ISO-8859-1 encoded.
   HeaderField(String name, String value) {
-    this(name.getBytes(ISO_8859_1), value.getBytes(ISO_8859_1));
+    this(name, value, value);
+  }
+
+  // This constructor can only be used if name and value are ISO-8859-1 encoded.
+  // Primarily used to initialize the static stable
+  HeaderField(String name, String value, Object valueAnnotation) {
+    this(name.getBytes(ISO_8859_1), name, value.getBytes(ISO_8859_1));
+    this.valueAnnotation = valueAnnotation;
   }
 
-  HeaderField(byte[] name, byte[] value) {
+  HeaderField(byte[] name, String nameString, byte[] value, Object valueAnnotation) {
```

Comment:
would prefer to have called context initialize cached values dynamically during addHeader()